system/Filters/Filters.php

```diff
@@ -494,10 +494,22 @@ protected function processMethods()
             return;
         }
 
-        // Request method won't be set for CLI-based requests
-        $method = strtolower($this->request->getMethod()) ?? 'cli';
+        $method = $this->request->getMethod();
+
+        $found = false;
 
         if (array_key_exists($method, $this->config->methods)) {
+            $found = true;
+        }
+        // Checks lowercase HTTP method for backward compatibility.
+        // @deprecated 4.5.0
+        // @TODO remove this in the future.
+        elseif (array_key_exists(strtolower($method), $this->config->methods)) {
+            $found  = true;
+            $method = strtolower($method);
+        }
```

Comment:
Should we trigger an `E_USER_DEPRECATED` here when this elseif branch is executed so that users will be aware?

Reply:
I like that

Reply:
Done.